### PR TITLE
Add Positive Definate Woodbury

### PR DESCRIPTION
**Pull request by oxinabox**
First commit.
First commit is just the old close-source code.

Only real change is to use ChainRulesCore rather than Zygote to make non-differnetiable.
Some updates to the tests.

Particular fiddly was realising that their was a breaking change in Distributions 0.23.8->0.23.9 that snuck though.
https://github.com/JuliaStats/Distributions.jl/pull/1160

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -1,21 +1,27 @@
...
 authors = ["Invenia Technical Computing"]
-version = "2.0.0"
+version = "2.1.0"
 
 [deps]
+ChainRulesCore = "d360d2e6-b24c-11e9-a2a3-2a2ae2dbcce4"
 LinearAlgebra = "37e2e46d-f89d-539d-b4ee-838fcccc9c8e"
 PDMats = "90014a1f-27ba-587c-ab20-58faa44d9150"
 
 [compat]
```
Comment (rofinn) [inline Project.toml:11]: Should `ChainRulesCore` have a compat?

File: test/woodbury_pd_mat.jl [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        @test isapprox(
+            logpdf(Distributions.GenericMvTDist(α, m, W), x),
+            logpdf(Distributions.GenericMvTDist(α, m, W_dense), x),
+        )
+
+        test_ad(randn(), α, m, A, D, S, x) do α, m, A, D, S, x
+            W = WoodburyPDMat(A, D, S)
+            return logpdf(Distributions.GenericMvTDist(α, m, W), x)
+        end
+    end
```
Comment (rofinn) [inline test/woodbury_pd_mat.jl:64]: Don't we also want to reuse our tests cases from `testutils.jl`?

**Review by rofinn — APPROVED** [review]
LGTM, just a couple minor suggestions.

File: Project.toml (hunk 1 above)
Comment (oxinabox) [reply Project.toml:11]: yes, it should

File: test/woodbury_pd_mat.jl (hunk 2 above)
Comment (oxinabox) [reply test/woodbury_pd_mat.jl:64]: That is issue #4 , I didn't want to add anything in this PR that wasn't in the close source code, except dropping the very heavy Zygote dependency

Comment (rofinn) [reply test/woodbury_pd_mat.jl:64]: Ahh, okay, thanks.

**Comment by oxinabox** [comment]
CI is only failing on nightly, because Zygote  doesn't work on nightly

**Pull request merged**